packages/encrypted-blockstore/src/connection.ts

```diff
@@ -48,17 +48,19 @@ export abstract class Connection {
     await this.loader!.remoteWAL?._process()
   }
 
-  connect({ loader }: { loader: Loader | undefined }) {
-    if (!loader) throw new Error('loader is required')
-    this.connectMeta({ loader })
-    this.connectStorage({ loader })
+  async connect({ loader }: { loader: Loader | undefined }) {
+    if (!loader) throw new Error('connect: loader is required')
+    Promise.all([
+      this.connectMeta({ loader }),
+      this.connectStorage({ loader })
+    ])
   }
 
-  connectMeta({ loader }: { loader: Loader | undefined}) {
-    if (!loader) throw new Error('loader is required')
+  async connectMeta({ loader }: { loader: Loader | undefined }) {
+    if (!loader) throw new Error('connectMeta: loader is required')
     this.loader = loader
     this.taskManager = new TaskManager(loader)
-    this.onConnect()
+    await this.onConnect()
```

Comment:
lets watch ths for does it change expected behavior in other connectors. I think there are places that expect this to be syncronous.

Reply:
we have to change it in any ways, i'm aware it could breaks things -- but it's impossible to sync a async connect function in javascript so the await is need and no problem on the first sight the onConnect is also async, and there is no problem if onConnect does not return a Promise modern javascript can handle that anyways